packages/components/src/ui/context/polymorphic-component.ts

```diff
@@ -3,16 +3,15 @@
  */
 // eslint-disable-next-line no-restricted-imports
 import type * as React from 'react';
-import type { As, RenderProp, ExtractHTMLAttributes } from 'reakit-utils/types';
+import type { As } from 'reakit-utils/types';
 import type { Interpolation } from 'create-emotion';
 
 /**
  * Based on https://github.com/reakit/reakit/blob/master/packages/reakit-utils/src/types.ts
  */
 export type ViewOwnProps< P, T extends As > = P &
-	Omit< React.ComponentPropsWithRef< T >, 'as' | keyof P > & {
+	Omit< React.ComponentPropsWithRef< T >, 'as' | keyof P | 'children' > & {
```

Comment:
Maybe we should add a comment noting why we're omitting `children` in this case? Just something saying that `ComponentPropsWithRef` will add it implicitly and we want to make sure it's explicit when needed.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/31817/commits/8011e6d835f75c1102a1147def7ef254f75f1217